server/api/src/test/java/org/hawkular/apm/server/api/model/zipkin/SpanTest.java

```diff
@@ -55,4 +67,73 @@ public void testPartialUrl() throws MalformedURLException {
         assertEquals(url.getPath(), result.getPath());
     }
 
+    @Test
+    public void testSerialization() throws IOException, ClassNotFoundException {
+        BinaryAnnotation stringAnnotation = new BinaryAnnotation();
+        stringAnnotation.setKey("http.method");
+        stringAnnotation.setValue("GET");
+        stringAnnotation.setEndpoint(createEndpoint("bonjour", "127.0.0.2", 8090));
+        stringAnnotation.setType(AnnotationType.STRING);
+
+        BinaryAnnotation numberAnnotation = new BinaryAnnotation();
+        numberAnnotation.setKey("foo");
+        numberAnnotation.setValue("265.66");
+        numberAnnotation.setEndpoint(createEndpoint("bonjour2", "127.0.1.2", 8090));
+        numberAnnotation.setType(AnnotationType.DOUBLE);
+
+        Span span = new Span(Arrays.asList(stringAnnotation, numberAnnotation));
+        span.setAnnotations(serverAnnotations());
+
+        Span deserializedSpan = (Span)deserialize(serialize(span));
+
+        Assert.assertEquals(span.binaryAnnotationMapping(), deserializedSpan.binaryAnnotationMapping());
```

Comment:
The assertion should be checking equivalent of 'span' against 'deserializedSpan' - to make sure both objects are the same - not just some of the fields.